daemon/api_sideload_n_try.go

```diff
@@ -285,9 +336,21 @@ func sideloadSnap(st *state.State, snapFile *uploadedSnap, flags sideloadFlags)
 		return nil, InternalError(err.Error())
 	}
 
+	// These two are filled only for components
+	var compInfo *snap.ComponentInfo
+	var snapInfo *snap.Info
+
 	sideInfo, apiErr := readSideInfo(st, snapFile.tmpPath, snapFile.filename, flags, deviceCtx.Model())
 	if apiErr != nil {
-		return nil, apiErr
+		// try to load as a component
+		var compErr *apiError
+		compInfo, snapInfo, compErr = readComponentInfo(st, snapFile.tmpPath, instanceName, flags)
```

Comment:
I think ultimately we want a read function that detects from the assertions or through looking at what is in the meta dir whether we deal with a component or a snap? that makes the most sense when we add code to find matching assertions if they exist, it would be good to leave a TODO about this though

Reply:
I've added the TODO